iree/compiler/Conversion/Common/PadLinalgOps.cpp

```diff
@@ -0,0 +1,151 @@
+// Copyright 2021 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "mlir/Dialect/Linalg/IR/LinalgOps.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+namespace mlir {
+namespace iree_compiler {
+
+namespace {
+/// A pattern to pad staticly shaped matmul operands to the next integer
+/// multiple of padSize.
+class PadMatmulOp : public OpRewritePattern<linalg::MatmulOp> {
+ public:
+  PadMatmulOp(MLIRContext *context, int size, PatternBenefit benefit = 1)
+      : OpRewritePattern<linalg::MatmulOp>(context, benefit),
+        paddingSize(size) {}
+
+  LogicalResult matchAndRewrite(linalg::MatmulOp matmulOp,
+                                PatternRewriter &rewriter) const override {
+    auto loc = matmulOp.getLoc();
+    auto lhs = matmulOp.inputs()[0];
+    auto rhs = matmulOp.inputs()[1];
+    auto result = matmulOp.outputs()[0];
+
+    if (lhs.getDefiningOp<linalg::PadTensorOp>() ||
+        rhs.getDefiningOp<linalg::PadTensorOp>()) {
+      return failure();
+    }
+
+    auto lhsShapeType = lhs.getType().cast<ShapedType>();
```

Comment:
Make this `RankedTensorType`. Also if you cast and the type is not a `ShapedType` it will assert. So either use `dyn_cast` here, or drop the `if (!lhsShapedType || !rhsShapedType)` below.